assets/src/dashboard/components/storyAnimation/provider.js

```diff
@@ -49,24 +55,43 @@ const createOnFinishPromise = (animation) => {
 };
 
 function Provider({ animations, children, onWAAPIFinish }) {
-  const animationGeneratorMap = useMemo(() => {
-    return (animations || []).reduce((map, animation) => {
-      const { targets, type, ...args } = animation;
+  const { map: animationPartsMap, set: animationTypesSet } = useMemo(() => {
+    return (animations || []).reduce(
+      ({ map, set }, animation) => {
+        const { targets, type, ...args } = animation;
+
+        (targets || []).forEach((t) => {
+          const generatedParts = map.get(t) || [];
+          map.set(t, [...generatedParts, AnimationPart(type, args)]);
```

Comment:
In order to call the `AMPAnimation` and `AMPAnimator` _separately_, I had to generate the `AnimationPart` immediately instead of passing in a function that would generate the animation part later.  This way, on load, the provider is full of generated and stable animation parts that can be accessed as needed and won't get regenerated each time. 

@littlemilkstudio This is the biggest change to the provider that I made, is this okay?  Does this break a future use you were expecting to leverage or does it couple things too much for your liking?  Feel free to push back if this really messes with things.